src/simulation/dynamics/spinningBodies/spinningBodiesNDOF/spinningBodyNDOFStateEffector.h

```diff
@@ -0,0 +1,215 @@
+    /** setter for `r_ScS_S` property */
+    void setR_ScS_S(Eigen::Vector3d r_ScS_S) {this->r_ScS_S = r_ScS_S;};
+    /** setter for `ISPntSc_S` property */
+    void setISPntSc_S(const Eigen::Matrix3d& ISPntSc_S) {this->ISPntSc_S = ISPntSc_S;};
+    /** setter for `sHat_S` property */
+    void setSHat_S(Eigen::Vector3d sHat_S);
+    /** setter for `dcm_S0P` property */
+    void setDCM_S0P(const Eigen::Matrix3d& dcm_S0P) {this->dcm_S0P = dcm_S0P;};
+    /** setter for `k` property */
+    void setK(double k);
+    /** setter for `c` property */
+    void setC(double c);
+    /** setter for `thetaInit` property */
+    void setThetaInit(double thetaInit) {this->thetaInit = thetaInit;};
+    /** setter for `thetaDotInit` property */
+    void setThetaDotInit(double thetaDotInit) {this->thetaDotInit = thetaDotInit;};
+
+    /** getter for `mass` property */
+    double getMass() const {return this->mass;};
+    /** getter for `r_SP_P` property */
+    Eigen::Vector3d getR_SP_P() const {return this->r_SP_P;};
+    /** getter for `r_ScS_S` property */
+    Eigen::Vector3d getR_ScS_S() const {return this->r_ScS_S;};
+    /** getter for `ISPntSc_S` property */
+    Eigen::Matrix3d setISPntSc_S() const {return this->ISPntSc_S;};
+    /** getter for `sHat_S` property */
+    Eigen::Vector3d getSHat_S() const {return this->sHat_S;};
+    /** getter for `dcm_S0P` property */
+    Eigen::Matrix3d getDCM_S0P() const {return this->dcm_S0P;};
+    /** getter for `k` property */
+    double getK() const {return this->k;};
+    /** getter for `c` property */
+    double getC() const {return this->c;};
+    /** getter for `thetaInit` property */
+    double getThetaInit() const {return this->thetaInit;};
+    /** getter for `thetaDotInit` property */
+    double getThetaDotInit() const {return this->thetaDotInit;};
+
+private:
+    // Make sure the effector can access each spinning body's private and protected members
+    friend class SpinningBodyNDOFStateEffector;
+
+    double thetaInit = 0.0;           //!< [rad] initial spinning body angle
+    double thetaDotInit = 0.0;        //!< [rad/s] initial spinning body angle rate
+    double k = 0.0;                   //!< [N-m/rad] torsional spring constant
+    double c = 0.0;                   //!< [N-m-s/rad] rotational damping coefficient
+    double mass = 1.0;                //!< [kg] spinning body mass
+    Eigen::Vector3d r_SP_P = Eigen::Vector3d::Zero();            //!< [m] vector pointing from parent frame P origin to spinning frame S origin in P frame components
+    Eigen::Vector3d r_ScS_S = Eigen::Vector3d::Zero();           //!< [m] vector pointing from spinning frame S origin to point Sc (center of mass of the spinner) in S frame components
+    Eigen::Vector3d sHat_S = {1.0, 0.0, 0.0};           //!< -- spinning axis in S frame components
+    Eigen::Matrix3d dcm_S0P = Eigen::Matrix3d::Identity();       //!< -- DCM from the parent frame to the S0 frame (S frame for theta=0)
```

Comment:
If there are no units, don't include `--`.  Not needed, and it causes the HTML files to have odd `&#8212;` text instead of `--`.  Please remove `--` for all unitless variable definitions.

Reply:
Done.